lib/bson/binary.js

```diff
@@ -10,7 +10,7 @@ function convert(integer) {
 }
 
 function toExtendedJSON(obj) {
-  var base64String = obj.buffer.toString('base64');
+  var base64String = Buffer.from(obj.buffer).toString('base64');
```

Comment:
If `obj.buffer` is a `Buffer`, then `Buffer.from` will make a copy of it, which is unnecessary here. Can we use `Buffer.isBuffer` to check if it is a buffer before using `Buffer.from`?